### PR TITLE
fixed some dead link and typo

### PR DESCRIPTION
**Pull request by techyang**
## What is the purpose of the change

fixed dead link online qos

## Brief changelog

fixed some dead link and typo

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo-website/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Test your code locally by running `docsite start`, and make sure it works as expected.
- [ ] Make sure no files under build directory is added.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: docs/en-us/user/references/protocol/rest.md [hunk 1]
```diff
@@ -0,0 +1,741 @@
...
+}
+
+if (RpcContext.getContext().getResponse() != null && RpcContext.getContext().getResponse() instanceof HttpServletResponse) {
+    System.out.println("Response object from RpcContext: " + RpcContext.getContext().getResponse());
+}
+```
+
+> 注意：为了保持协议的中立性，RpcContext.getRequest()和RpcContext.getResponse()返回的仅仅是一个Object类，而且可能为null。所以，你必须自己做null和类型的检查。
+
+> 注意：只有在设置server="jetty"或者server="tomcat"或者server="servlet"的时候，你才能通过以上方法正确的得到HttpServletRequest和HttpServletResponse，因为只有这几种server实现了servlet容器。
```
Comment (lixiaojiee) [inline docs/en-us/user/references/protocol/rest.md]: “只有在设置server="jetty"或者server="tomcat"或者server="servlet"的时候” 可替换为“只有在将 server 设置为 jetty、tomcat 或者 servlet 的时候，”

File: docs/en-us/user/references/protocol/rest.md [hunk 2]
```diff
@@ -0,0 +1,741 @@
...
+    @Path("register")
+    @Consumes({MediaType.APPLICATION_JSON})
+    public void registerUser(User user) {
+        // save the user
+        userRepository.save(user);
+    }	
+}
+```
+
+annotation的配置更简单更精确，经常也更便于维护（当然现代IDE都可以在xml中支持比如类名重构，所以就这里的特定用例而言，xml的维护性也很好）。而xml对代码的侵入性更小一些，尤其有利于动态修改配置，特别是比如你要针对单个服务配置连接超时时间、每客户端最大连接数、集群策略、权重等等。另外，特别对复杂应用或者模块来说，xml提供了一个中心点来涵盖的所有组件和配置，更一目了然，一般更便于项目长时期的维护。
```
Comment (lixiaojiee) [inline docs/en-us/user/references/protocol/rest.md]: annotation的配置更简单更精确，**通常**也更便于维护（当然现代IDE都可以在xml中支持比如类名重构，所以就这里的特定用例而言，xml的维护性也很好）。而xml对代码的侵入性更小一些，尤其有利于动态修改配置，特别是比如你要针对单个服务配置连接超时时间、每客户端最大连接数、集群策略、权重等等。另外，~~特别~~对复杂应用或者模块来说，xml提供了一个中心点来涵盖的所有组件和配置，更一目了然，一般更便于项目长时期的维护。

File: docs/en-us/user/references/protocol/rest.md [hunk 3]
```diff
@@ -0,0 +1,741 @@
+# rest://
```
[the remark this answers is not in the document]
Comment (techyang) [reply docs/en-us/user/references/protocol/rest.md]: The english rest.md  doc is missing,  so i copied the zh-cn file to this doc . Someone please translate it into english.

**Comment by techyang** [comment]
Please review this pr again.

**Comment by techyang** [comment]
Please review this pr again.

**Comment by lixiaojiee** [comment]
It looks good to me, thank you

File: docs/zh-cn/dev/principals/configuration.md [hunk 4]
```diff
@@ -30,7 +30,7 @@ Dubbo 新版本去掉了 dubbo.properties，因为该约定经常造成配置冲
 
 在 Dubbo 即存在描述配置，也有环境配置。一部分用 spring 的 schema 配置加载，一部分从 classpath 扫描 properties 配置加载。用户感觉非常不便，所以在新版本中进行了合并，统一放到 spring 的 schema 配置加载，也增加了配置的灵活性。 
 
-扩展配置，通常对配置的聚合要求比较高。因为产品需要发现第三方实现，将其加入产品内部。在 java 世界里，通常是约定在每个 jar 包下放一个指定文件加载，比如：eclipse 的 plugin.xml，struts2 的 struts-plugin.xml 等，这类配置可以考虑 java 标准的服务发现机制，即在 jar 包的 META-INF/services 下放置接口类全名文件，内容为每行一个实现类类名，就像 jdk 中的加密算法扩展，脚本引擎扩展，新的 JDBC 驱动等，都是采用这种方式。参见：[ServiceProvider 规范](http://download.oracle.com/javase/1.4.2/docs/guide/jar/jar.html#Service%20Provider)。
+扩展配置，通常对配置的聚合要求比较高。因为产品需要发现第三方实现，将其加入产品内部。在 java 世界里，通常是约定在每个 jar 包下放一个指定文件加载，比如：eclipse 的 plugin.xml，struts2 的 struts-plugin.xml 等，这类配置可以考虑 java 标准的服务发现机制，即在 jar 包的 META-INF/services 下放置接口类全名文件，内容为每行一个实现类类名，就像 jdk 中的加密算法扩展，脚本引擎扩展，新的 JDBC 驱动等，都是采用这种方式。参见：[ServiceProvider 规范](https://docs.oracle.com/javase/1.5.0/docs/guide/jar/jar.html#Service%20Provider)。
```
Comment (lixiaojiee) [inline docs/zh-cn/dev/principals/configuration.md]: Hi，刚才我在检查pr 245的时候，发现了同样的更改点. @Skqing 修改的链接是针对jdk 1.8 ,你这个是针对jdk 1.5 ,为了防止将来可能由于提交顺序问题被覆盖掉，你可以将这个地址改为https://docs.oracle.com/javase/tutorial/sound/SPI-intro.html吗？

Comment (techyang) [reply docs/zh-cn/dev/principals/configuration.md]: > Hi，刚才我在检查pr 245的时候，发现了同样的更改点. @Skqing 修改的链接是针对jdk 1.8 ,你这个是针对jdk 1.5 ,为了防止将来可能由于提交顺序问题被覆盖掉，你可以将这个地址改为[https://docs.oracle.com/javase/tutorial/sound/SPI-intro.html吗？](https://docs.oracle.com/javase/tutorial/sound/SPI-intro.html%E5%90%97%EF%BC%9F)

ok, already done. please review again.

**Comment by lixiaojiee** [comment]
It looks good to me, thank you! @ralf0131 This pr has been reviewed.

**Comment by ralf0131** [comment]
@lixiaojiee You can approve this pull request officially, rather than leaving a comment.

**Comment by lixiaojiee** [comment]
> @lixiaojiee You can approve this pull request officially, rather than leaving a comment.

done

**Pull request merged**